### PR TITLE
Fix the getHeaders method signatures

### PR DESCRIPTION
**Pull request by ScriptSmith**
In the readme, `getHeaders` is described as accepting and returning an array:

```
Array getHeaders( [Array userHeaders] )
```

The function doesn't return an array, and while it can technically accept one, the output it produces doesn't seem useful:

https://github.com/form-data/form-data/blob/905f173a3f785e8d312998e765634ee451ca5f42/lib/form_data.js#L293-L306

Here's how it's defined in [index.d.ds](https://github.com/form-data/form-data/blob/905f173a3f785e8d312998e765634ee451ca5f42/index.d.ts):

https://github.com/form-data/form-data/blob/905f173a3f785e8d312998e765634ee451ca5f42/index.d.ts#L14

https://github.com/form-data/form-data/blob/905f173a3f785e8d312998e765634ee451ca5f42/index.d.ts#L26-L29

This is more correct, but it says that `getHeaders` doesn't take any arguments, which is incorrect.

I've updated the readme to say that it takes and returns `Headers` as defined in `index.d.ts` (but perhaps it should be something more generic like `Object`). I've also updated the type definitions to include the `userHeaders` parameter.

Furthermore, line 300 in `getHeaders` seems redundant:

https://github.com/form-data/form-data/blob/905f173a3f785e8d312998e765634ee451ca5f42/lib/form_data.js#L300

It was [added in ebf47ef](
https://github.com/form-data/form-data/commit/ebf47ef605b6d1e6a730836234a460e080bf79d2#diff-121026c01234c71e2b3b316f7abc5afcR217-R230) because of an issue identified in codacity? / codacy. I'm not sure what problem this could be mitigating.

If it is redundant I'll add another commit to correct it.

### CODE REVIEW

**Comment by mapleeit** [comment]
Hi @ScriptSmith  Thank you for your patch and sorry for the late reply. (I missed this)

I'm not sure the type of input param is just 

```
interface Headers {
    [key: string]: any;
  }
```

I guess it may be `NodeJS.IncomingHttpHeaders`. Otherwise the `hasOwnProperty` is redundant.

@alexindigo Do you still remember the reason why adding `hasOwnProperty`? : ） https://github.com/form-data/form-data/commit/ebf47ef605b6d1e6a730836234a460e080bf79d2#diff-121026c01234c71e2b3b316f7abc5afcR217-R230

**Comment by ScriptSmith** [comment]
Good point, I think `OutgoingHttpHeaders` would be more appropriate given that form-data is for client-side applications. 

Regarding `hasOwnProperty`, I believe it's because it's trying to ignore inherited properties from the object's prototype chain. Maybe there are some packages that have really complicated header objects?

**Comment by ScriptSmith** [comment]
So how should `form-data` be interacting with the headers it's provided with? And how picky should it be?

I think it should just look at the object's keys, and not up the prototype chain. Therefore it makes sense to keep the `hasOwnProperty` check as it is.

I also think it shouldn't be picky about the types of header values, even though internally it only makes sense for them to be strings / numbers, because I think other users are relying on the non-standard `getHeaders` function to generate the `content-type` header:

https://github.com/form-data/form-data/blob/5c6bf52b25c77c96e16a9456847365feafb4d7ab/lib/form_data.js#L295-L297

Therefore I think it should just use the simple definition that's already included, rather than any of the header definitions in `http`

What do you think @mapleeit?

File: index.d.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ interface Options {
 declare class FormData extends stream.Readable {
   constructor(options?: Options);
   append(key: string, value: any, options?: FormData.AppendOptions | string): void;
-  getHeaders(): FormData.Headers;
+  getHeaders(userHeaders: FormData.Headers): FormData.Headers;
```
Comment (mapleeit) [inline index.d.ts]: `userHeaders?: FormData.Headers ` should be more accurate.

**Comment by mapleeit** [comment]
@ScriptSmith Agree with that. `FormData.Headers` could do. Although people may use `http. OutgoingHttpHeaders` param for most cases.

**Pull request merged**